### PR TITLE
Set up performance benchmark folder

### PR DESCRIPTION
**Pull request by gassmoeller**
This is an attempt to start a collection of performance benchmarks that should help users to compare ASPECTs performance and scaling to other clusters. This will help us collect data about ASPECTs scaling, and it will help users to make sure they set up their code correctly. Comments about the folder structure and about what kind of data should be collected are appreciated.

### CODE REVIEW

File: benchmarks/performance_benchmarks/spherical_shell_expensive_solver/HLRN_Berlin_aspect_2.0-pre/screen-output/output_1536_6_1 [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+-----------------------------------------------------------------------------
+The output directory <outputdir_1536_6_1/> provided in the input file appears not to exist.
+ASPECT will create it for you.
+-----------------------------------------------------------------------------
+
+
+Number of active cells: 25,165,824 (on 7 levels)
+Number of degrees of freedom: 1,035,930,380 (606,340,614+25,362,690+202,113,538+202,113,538)
```
Comment (bangerth) [inline benchmarks/performance_benchmarks/spherical_shell_expensive_solver/HLRN_Berlin_aspect_2.0-pre/screen-output/output_1536_6_1]: !!!!

File: benchmarks/performance_benchmarks/spherical_shell_expensive_solver/HLRN_Berlin_aspect_2.0-pre/screen-output/output_1536_6_1 [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Application 4320742 exit signals: Killed
+Application 4320742 resources: utime ~128s, stime ~130s, Rss ~46688, inblocks ~715904, outblocks ~2137728
```
Comment (bangerth) [inline benchmarks/performance_benchmarks/spherical_shell_expensive_solver/HLRN_Berlin_aspect_2.0-pre/screen-output/output_1536_6_1]: but of course then this happens...

**Comment by tjhei** [comment]
I vote for creating a separate repo for all this data, but we should probably discuss in a group...

**Comment by bangerth** [comment]
Would the wiki be a better place for it?

**Comment by gassmoeller** [comment]
A while ago I started something in the wiki and then newer quite followed up on it (see https://github.com/geodynamics/aspect/wiki/Compatible-systems). We might add links to scaling data for these clusters. I still vote for putting the parameter files into the repository, but I am fine putting the output data somewhere else.

**Comment by bangerth** [comment]
So what do we do about this?

**Comment by gassmoeller** [comment]
@naliboff wanted to contribute more data to this, so we should probable think about a solution. I still think the setups and instructions about how to run them, and how to compare with existing data should be part of the main repository, but I agree the data might go somewhere else. That might be a bit confusing, but what about we start a separate repo for the data and add it as submodule to this folder? This way people would not immediately have to get all that unrelated data (unless they do `git clone --recursive`), but it could still be reached from within ASPECT. That separate repo could even get a github pages website that presents the scaling results. Would that be reasonable?

**Comment by tjhei** [comment]
> Would that be reasonable?

I would very much prefer a separate repo for the data files, yes. I would be okay with adding a submodule, but I could also understand that it might be cleaner without. We can just link to the repo in the readme in that folder.

**Comment by bangerth** [comment]
Either way is fine by me. As is just putting everything into the main repo for now, and seeing how that works out. We don't need to engineer a solution that may turn out to be overkill later on.

**Comment by gassmoeller** [comment]
I have removed the data and visualization from this PR, and have opened a separate repository for it (see description in the README.md file). The new repository is currently named 
[gassmoeller/aspect-performance-statistics](https://github.com/gassmoeller/aspect-performance-statistics), but I am open to suggestions about name or location of that repository. Currently the parameter files of the benchmark cases are in both repos, how should we avoid getting out of sync? Do we just manually update both every time we update one of them?

**Comment by naliboff** [comment]
@gassmoeller - thanks for setting up the new repository. Shall we continue discussion about specific benchmarks and results here, in a new issue or on aspect-devel?

**Comment by gassmoeller** [comment]
Since this PR is ready for a review and potentially merge, lets move the discussion about additional test cases or results to another place. Both a new issue or the mailing list would  be fine with me.

**Comment by gassmoeller** [comment]
I think the separate repository works well enough. Lets close this unless we see any further use for it in the future.

**Pull request closed without merging**